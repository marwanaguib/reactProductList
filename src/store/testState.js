export default {
  products: {
    count: 7,
    pageIndex: 0,
    departments: [
      { id: 1, name: "Electronics" },
      { id: 2, name: "Toys" },
      { id: 3, name: "Cloths" },
    ],
    filters: {},
    items: [
      { id: 1, name: "Product1", price: "100", promo: "10%", isActive: 1 },
      { id: 2, name: "Product2", price: "110", promo: "10%", isActive: 0 },
      { id: 3, name: "Product3", price: "80", promo: "10%", isActive: 1 },
      { id: 4, name: "Product4", price: "100", promo: "10%", isActive: 1 },
      { id: 5, name: "Product5", price: "110", promo: "10%", isActive: 1 },
      { id: 6, name: "Product6", price: "100", promo: "10%", isActive: 1 },
      { id: 7, name: "Product7", price: "100", promo: "10%", isActive: 1 },
    ],
  },
  shared: {
    errorMsg: "",
    isLoading: false,
  },
};
