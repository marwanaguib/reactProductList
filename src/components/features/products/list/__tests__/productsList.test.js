import React from "react";
import { shallow } from "enzyme";

import List from "../index";
import testState from "../../../../../store/testState";

const props = {
  products: testState.products.items,
};

describe("productsList", () => {
  it("render List to match snapshot", () => {
    const componetWrapper = shallow(<List {...props} />);
    expect(componetWrapper).toMatchSnapshot();
  });

  it("List is empty is products array empty", () => {
    const componetWrapper = shallow(<List {...{ ...props, products: [] }} />);
    expect(componetWrapper.find("h4").text()).toBe("No Products Found");
    expect(componetWrapper).toMatchSnapshot();
  });
});
