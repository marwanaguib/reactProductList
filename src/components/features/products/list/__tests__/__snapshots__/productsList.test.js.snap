// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`productsList List is empty is products array empty 1`] = `
<styled.div>
  <h4>
    No Products Found
  </h4>
</styled.div>
`;

exports[`productsList render List to match snapshot 1`] = `
<styled.div>
  <styled.div
    key="1"
  >
    <styled.div
      justify="space-between"
    >
      <styled.h2>
        Product1
      </styled.h2>
      <div>
        <s>
          100
          $
        </s>
      </div>
    </styled.div>
    <styled.div
      justify="flex-end"
    >
      <styled.div>
        ↓ 
        10%
      </styled.div>
      <div>
        90
        $
      </div>
    </styled.div>
  </styled.div>
  <styled.div
    key="2"
  >
    <styled.div
      justify="space-between"
    >
      <styled.h2>
        Product2
      </styled.h2>
      <div>
        110
        $
      </div>
    </styled.div>
  </styled.div>
  <styled.div
    key="3"
  >
    <styled.div
      justify="space-between"
    >
      <styled.h2>
        Product3
      </styled.h2>
      <div>
        <s>
          80
          $
        </s>
      </div>
    </styled.div>
    <styled.div
      justify="flex-end"
    >
      <styled.div>
        ↓ 
        10%
      </styled.div>
      <div>
        72
        $
      </div>
    </styled.div>
  </styled.div>
  <styled.div
    key="4"
  >
    <styled.div
      justify="space-between"
    >
      <styled.h2>
        Product4
      </styled.h2>
      <div>
        <s>
          100
          $
        </s>
      </div>
    </styled.div>
    <styled.div
      justify="flex-end"
    >
      <styled.div>
        ↓ 
        10%
      </styled.div>
      <div>
        90
        $
      </div>
    </styled.div>
  </styled.div>
  <styled.div
    key="5"
  >
    <styled.div
      justify="space-between"
    >
      <styled.h2>
        Product5
      </styled.h2>
      <div>
        <s>
          110
          $
        </s>
      </div>
    </styled.div>
    <styled.div
      justify="flex-end"
    >
      <styled.div>
        ↓ 
        10%
      </styled.div>
      <div>
        99
        $
      </div>
    </styled.div>
  </styled.div>
  <styled.div
    key="6"
  >
    <styled.div
      justify="space-between"
    >
      <styled.h2>
        Product6
      </styled.h2>
      <div>
        <s>
          100
          $
        </s>
      </div>
    </styled.div>
    <styled.div
      justify="flex-end"
    >
      <styled.div>
        ↓ 
        10%
      </styled.div>
      <div>
        90
        $
      </div>
    </styled.div>
  </styled.div>
  <styled.div
    key="7"
  >
    <styled.div
      justify="space-between"
    >
      <styled.h2>
        Product7
      </styled.h2>
      <div>
        <s>
          100
          $
        </s>
      </div>
    </styled.div>
    <styled.div
      justify="flex-end"
    >
      <styled.div>
        ↓ 
        10%
      </styled.div>
      <div>
        90
        $
      </div>
    </styled.div>
  </styled.div>
</styled.div>
`;
